test/e2e/csc_tests.go

```diff
@@ -131,3 +151,83 @@ func cscWithNonExistingTargetNamespace(t *testing.T, f *test.Framework, ctx *tes
 
 	return nil
 }
+
+func cscWithNonExistingPackage(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// Create a new catalogsourceconfig with a non-existing Package
+	nonExistingPackageCSC := &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		}, ObjectMeta: metav1.ObjectMeta{
+			Name:      nonExistingPackageCSCName,
+			Namespace: namespace,
+		},
+		Spec: operator.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        packageName,
+		}}
+	err = createRuntimeObject(f, ctx, nonExistingPackageCSC)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig with a non-existing package
+	resultCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, nonExistingPackageCSCName)
+	if err != nil {
+		return err
+	}
+
+	// Check if the catalogsourceconfig phase and message are the expected values
+	expectedPhase := "Configuring"
```

Comment:
Should the phase be "Failed" in this case, since the message mentions the package wasn't found? If it is expected to be "Configuring" does the object eventually resolve to "Failed" or does it get stuck?

Reply:
@SamiSousa it doesn't get set to failed because it retries until it could succeed. consider the case where a new package was specified but just wasn't picked up by the operator source yet through an update. in this case, it would be eventually consistent since it retries until it can succeed. otherwise a failed state requires manual intervention.